### PR TITLE
Fixes most eye color effects not working

### PR DESCRIPTION
**Pull request by SmArtKar**
## About The Pull Request

Whoever refactored eye color to be separated forgot about status effects and it resulted in those not working. I just added a couple eye refresh() proc calls which should fix the issue. Also fixed poor refresh and insert code which wasnt working as well.

## Changelog

:cl:
fix: Fixed most eye color effects not working
/:cl:

### CODE REVIEW

**Review by ZephyrTFA — APPROVED** [review]

File: code/modules/mining/lavaland/megafauna_loot.dm [hunk 1]
```diff
@@ -728,6 +728,10 @@
 			consumer.eye_color_left = "#FEE5A3"
 			consumer.eye_color_right = "#FEE5A3"
 			consumer.set_species(/datum/species/lizard)
+			var/obj/item/organ/eyes/eyes = consumer.getorgan(/obj/item/organ/eyes)
+			if(eyes)
+				eyes.refresh()
```
Comment (MrMelbert) [inline code/modules/mining/lavaland/megafauna_loot.dm:733]: Don't need this anymore?

File: code/game/objects/structures/mirror.dm [hunk 2]
```diff
@@ -263,6 +263,9 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/structure/mirror, 28)
 				amazed_human.eye_color_right = sanitize_hexcolor(new_eye_color)
 				amazed_human.dna.update_ui_block(DNA_EYE_COLOR_LEFT_BLOCK)
 				amazed_human.dna.update_ui_block(DNA_EYE_COLOR_RIGHT_BLOCK)
+				var/obj/item/organ/eyes/eyes = amazed_human.getorgan(/obj/item/organ/eyes)
+				if(eyes)
+					eyes.refresh()
```
Comment (MrMelbert) [inline code/game/objects/structures/mirror.dm:268]: Don't need this anymore?

File: code/modules/mob/living/carbon/human/species_types/dullahan.dm [hunk 3]
```diff
@@ -49,6 +49,9 @@
 			head.eyes = new /obj/item/organ/eyes(head)
 			head.eyes.eye_color_left = human.eye_color_left
 			head.eyes.eye_color_right = human.eye_color_right
+			var/obj/item/organ/eyes/eyes = human.getorgan(/obj/item/organ/eyes)
+			if(eyes)
+				eyes.refresh()
```
Comment (MrMelbert) [inline code/modules/mob/living/carbon/human/species_types/dullahan.dm:54]: Update body instead?

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/modules/mining/lavaland/megafauna_loot.dm [hunk 4]
```diff
@@ -728,6 +728,7 @@
 			consumer.eye_color_left = "#FEE5A3"
 			consumer.eye_color_right = "#FEE5A3"
 			consumer.set_species(/datum/species/lizard)
+			consumer.update_body()
```
Comment (JohnFulpWillard) [inline code/modules/mining/lavaland/megafauna_loot.dm:731]: is this needed anymore?

Comment (SmArtKar) [reply code/modules/mining/lavaland/megafauna_loot.dm:731]: Yes? I moved eye refresh into update_body so you'll need to call it.

Comment (JohnFulpWillard) [reply code/modules/mining/lavaland/megafauna_loot.dm:731]: set_species wouldn't call this for us?

Comment (SmArtKar) [reply code/modules/mining/lavaland/megafauna_loot.dm:731]: oh right

**Pull request merged**